packages/workers-shared/asset-worker/tests/handler.test.ts

```diff
@@ -1251,4 +1251,169 @@ describe("[Asset Worker] `canFetch`", () => {
 			)
 		).toBeTruthy();
 	});
+
+	describe("assets_sec_fetch_mode_navigate_header_prefers_asset_serving", async () => {
+		const exists = (pathname: string) => {
+			if (["/404.html", "/index.html", "/foo.html"].includes(pathname)) {
+				return "some-etag";
+			}
+
+			return null;
+		};
+
+		interface CompatibilityOptions {
+			compatibilityDate?: string;
+			compatibilityFlags?: string[];
+		}
+
+		const compatibilityOptionsModes: [CompatibilityOptions, boolean][] = [
+			[{}, false],
+			[
+				{
+					compatibilityFlags: [
+						"assets_sec_fetch_mode_navigate_header_prefers_asset_serving",
+					],
+				},
+				true,
+			],
+			[
+				{
+					compatibilityFlags: [
+						"assets_not_found_handling_prefers_asset_worker",
+					],
+				},
+				false,
+			],
+			// TODO
```

Comment:
How should `flagIsEnabled` work when both are present? This maybe can't happen in practice if we do educate workerd about these flags and rely on it for validation? I think that's probably going to reject mutually exclusive flags. Lets.... blow up? Idk. Idc.